### PR TITLE
Adding lead_time to torch DeepAR and trivial predictor

### PR DESCRIPTION
**Pull request by raffienficiaud**
*Issue #1861 (related but not solving)*

*Description of changes:*

* adding lead_time to DeepAR pytorch implementation. 
* adding lead_time to trivial/`ConstantValuePredictor`

I am pretty new to GluonTS and it is likely I am missing some parts for implementing this properly. Those involve my current understanding of the framework as well as the DeepAR model (as per [this comment](https://github.com/awslabs/gluon-ts/issues/1861#issuecomment-1040102782)).
The changes "just work" on my setup. 

Note: I am not able to run `mypy` as the number of failures is high and they are unrelated to the changes I am making. Same for `pytest`. `black` has be run over the changes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by lostella** [comment]
Thanks for starting this! (And for the related deep dive in #1936!)

> The changes "just work" on my setup.

I understand the changes to the splitters, which make room for the additional `lead_time` time point at the end when iterating and splitting the data. At a high level those appear to be the required ones.

However, for DeepAR, I'm not sure that the right thing is happening under the hood: thinking about how the underlying RNN is used, what I mentioned in [my comment](https://github.com/awslabs/gluon-ts/issues/1861#issuecomment-1040102782) should translate into the DeepAR module doing the following:
* have a `lead_time` parameter too
* the `forward` method producing `lead_time + prediction_length` many samples (along the time axis), only to chop-off the first `lead_time`
* related changes to the lightning module used for training

Given this is it's not a trivial change, I think it would be nice to have some evidence showing that the estimator and module are doing the right thing with `lead_time` (for training and prediction, respectively). A simple example with some artificial, periodic data could be enough maybe.

Note that this results in the model doing all the work for the entire `lead_time + prediction_length` range, only discarding part of it. Arguably, a simpler way to get the same result is to leave the code as it is, set the estimator's `prediction_length` to `lead_time + prediction_length` when constructing it, and just post-process the forecasts that come out of the model (ignore their initial `lead_time` length).
